Remotion/ObjectBinding/Web.Test/IndividualControlTests/BocListUserControl.ascx.cs

```diff
@@ -354,6 +353,18 @@ override protected void OnLoad (EventArgs e)
   protected override void OnPreRender (EventArgs e)
   {
     base.OnPreRender(e);
+
+    var haveValidationErrors = (bool?)Session[c_haveValidationErrors] ?? false;
```

Comment:
This shouldn't be in the session. The session applies to all pages & page reload and would trigger sideeffects. You can put this into ViewState. There's examples for that in the test pages.

Actually, the checkbox should already preserve it's checkstate between postbacks via it's ViewState/ControlState so you don't need to do anything except read the value, which reduces the required code to `if (CheckBox.Checked) CreateValidationErrors();`